### PR TITLE
 improve navigation and bottom sheet 

### LINKED ISSUE #15265: Bottom sheet sometimes gets stuck when opening in chat view (Android)
# Bug Report

## Problem 

The bottom sheet on Android devices sometimes gets stuck just starting to open in this state:

Screenshot 2023-03-06 at 09 19 36

Here is logfile for the screenshot above. Event should be somewhere at the very beginning of log history. 
[Status-debug-logs (17).zip](https://github.com/status-im/status-mobile/files/10896096/Status-debug-logs.17.zip)

In the slow-mo video, you can see, that the animation is a bit laggy at this moment, and for some reason, the bottom sheet remains at the point of slowing down. 

https://user-images.githubusercontent.com/67952253/223055560-99717998-54aa-4fca-bedd-178565b56aae.mp4

### Reproduction

The exact steps to reproduce, unfortunately, have not yet been found. 

### Additional Information

- Status version: nightly 6/03
- Operating System: Android

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #15346: Incorrect height of the composer on Android
# Bug Report

## Problem 

The bottom padding is too small in the composer and bottom sheets on Android.

Screenshot 2023-03-14 at 17 12 31

---

Screenshot 2023-03-14 at 17 10 08

---

Designs:

Screenshot 2023-03-14 at 17 10 22

Screenshot 2023-03-14 at 17 16 47

### Reproduction

1. Open any chat / Tap the `Plus` button on the chat screen

### Additional Information

- Status version: nightly 14/03
- Operating System: Android
- Devices: Huawei P40 Lite, Pixel 2

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #15410: URL input field is overlapped by status bar on Browser screen
**Steps:**

1. Open Status
2. Go to Browser tab
3. Pay attention at URL input field location

**Actual result:** URL input field is located to high and overlapped by status bar on Browser screen

photo_2023-03-20 11 59 54

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #15438: Bottom sheet component expands too much on Android
cc @Parveshdhull 

Tested on Android 12, Samsung Galaxy A52

Below example demonstrates "Remove profile" bottom sheet behaviour. 

**Expected result:** 

Onboarding for Mobile – Figma 2023-03-20 16-46-30

**Actual result:** 

photo_2023-03-20 16 49 06
### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #15457: Migrate "New chat (contacts list)"  bottom-sheet to bottom-sheet screen

### LINKED ISSUE #15458: Hiding bottom sheet (component) does not animate
Calling `:bottom-sheet/hide` does not animate hiding the bottom sheet

Also, when calling `bottom-sheet/hide` to navigate to another screen, the bottom-sheet probably should animate a little faster (compared to the opening animation).

### PR DESCRIPTION
**Pull request by flexsurfer**
- Navigation has been simplified, now **top inset** is false by default, so no need to specify it for each screen, same for **topBar**
- Bottom sheet is now part of the navigation, so when navigate to another screen from bottom sheet, no need to close it, it will be closed by navigation
- Navigation namespace has been refactored and simplified 
- insets have been moved under options
- New option **:sheet?** for modals with bottom sheet UI

```clojure
{:name      :photo-selector
 :options   {:sheet? true}
 :component photo-selector/photo-selector}
```

**For reviewers: lots of changes are just moving stuff or renaming, so i add comments to files that have real changes, most of them in `navigation` namespace and in `common/bottom-sheet`**

fixes: https://github.com/status-im/status-mobile/issues/15265
fixes: https://github.com/status-im/status-mobile/issues/15458
fixes: https://github.com/status-im/status-mobile/issues/15457
fixes: https://github.com/status-im/status-mobile/issues/15438
fixes: https://github.com/status-im/status-mobile/issues/15346
fixes: https://github.com/status-im/status-mobile/issues/15410

Known issues:

- when closing bottom sheet , status bar has wrong theme on home screens
- when closing bottom sheet, navigation bar on android has wrong theme

**For QA: changes in this PR only apply to the bottom sheet with the new design, old bottom sheets still have all issues**

### CODE REVIEW

File: src/status_im/bottom_sheet/events.cljs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+(ns status-im.bottom-sheet.events
```
Comment (flexsurfer) [inline src/status_im/bottom_sheet/events.cljs:1]: old sheet just moved (no changes)

File: src/status_im/bottom_sheet/styles.cljs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+(ns status-im.bottom-sheet.styles
```
Comment (flexsurfer) [inline src/status_im/bottom_sheet/styles.cljs:1]: old sheet just moved (no changes)

File: src/status_im/bottom_sheet/view.cljs [hunk 3]
```diff
@@ -0,0 +1,240 @@
+(ns status-im.bottom-sheet.view
```
Comment (flexsurfer) [inline src/status_im/bottom_sheet/view.cljs:1]: old sheet just moved (no changes)

File: src/status_im2/common/bottom_sheet/view.cljs [hunk 4]
```diff
@@ -1,240 +1,83 @@
 (ns status-im2.common.bottom-sheet.view
-  (:require [oops.core :refer [oget]]
-            [quo.react :as react]
-            [status-im2.common.bottom-sheet.styles :as styles]
-            [re-frame.core :as re-frame]
-            [react-native.background-timer :as timer]
+  (:require [utils.re-frame :as rf]
```
Comment (flexsurfer) [inline src/status_im2/common/bottom_sheet/view.cljs:2]: new simplified bottom sheet

File: src/status_im2/navigation/events.cljs [hunk 5]
```diff
@@ -10,33 +10,43 @@
     (seq screen-params)
```
Comment (flexsurfer) [inline src/status_im2/navigation/events.cljs:10]: refactoring

File: src/status_im2/navigation/core.cljs [hunk 6]
```diff
@@ -1,88 +1,110 @@
 (ns status-im2.navigation.core
```
Comment (flexsurfer) [inline src/status_im2/navigation/core.cljs:1]: refactoring

File: src/status_im2/navigation/options.cljs [hunk 7]
```diff
@@ -0,0 +1,110 @@
+(ns status-im2.navigation.options
```
Comment (flexsurfer) [inline src/status_im2/navigation/options.cljs:1]: new namespace with options

File: src/status_im2/navigation/roots.cljs [hunk 8]
```diff
@@ -1,78 +1,26 @@
 (ns status-im2.navigation.roots
```
Comment (flexsurfer) [inline src/status_im2/navigation/roots.cljs:1]: refactoring

File: src/status_im2/navigation/options.cljs [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+  (:require [quo2.foundations.colors :as colors]
+            [react-native.platform :as platform]))
+
+(defn deafult-options
```
Comment (ilmotta) [inline src/status_im2/navigation/options.cljs]: Typo

**Review by ilmotta — APPROVED** [review]
Impressive work @flexsurfer :mechanical_arm: 

I'm just not liking the creation of *old* and *new* stuff in the new namespaces. I thought we agreed to stop generating this kind of complexity and confusion. For instance, there are lots of places in the new code doing this `(rf/dispatch [:bottom-sheet/hide-old])`.

Since this is already a huge PR in terms of diff, my suggestion is to open a separate, follow-up PR that eliminates these `hide-old` stuff from the new namespaces, so that you can merge this one quicker in order to avoid conflicts. I'm just vouching for us to keep the new namespaces clean of the mess of duplicated abstractions :)

**Comment by ilmotta** [comment]
@flexsurfer, consider not merging with the emojis in the commit title, otherwise this can start to be acceptable in our git history. Just a reminder because I saw you merged a commit in the past with emojis.

**Comment by flexsurfer** [comment]
> I'm just not liking the creation of _old_ and _new_ stuff in the new namespaces. I thought we agreed to stop generating this kind of complexity and confusion. For instance, there are lots of places in the new code doing this `(rf/dispatch [:bottom-sheet/hide-old])`.

yeah good point, i'll take a look


edited: actually its really great point, in some places it's a wrong usage of it

File: src/status_im2/navigation/core.cljs [hunk 10]
```diff
@@ -1,88 +1,110 @@
...
   [comp]
-  (log/debug "NAVIGATE" comp)
   (let [{:keys [options]} (get views/screens comp)]
+    (dismiss-all-modals)
     (navigation/push
      (name @state/root-id)
      {:component {:id      comp
                   :name    comp
-                  :options (merge (status-bar-options)
+                  :options (merge (options/statusbar-and-navbar)
```
Comment (J-Son89) [inline src/status_im2/navigation/core.cljs:85]: Consider removing any option merging logic to a helper function in options.cljs.
As I see it we have 3 different levels of options.
Default options (defined in options.cljs), root level options (Defined on the root stack in roots.cljs) and screen options.
This function here should not have any options directly being set (e.g (if topBar options etc....)
Instead it should just do a nested merge respecting this 3 tier hierarchy for the different options. 


```
  {:component {:id      comp
                  :name    comp
                  :options (merge-options (options/default-root)
                                                              (roots/get-stack-options stack-id) ;;
                                                               options)
 ```
Where this merge-options function has to be created but better to have the logic abstracted outside this navigate function as we might need to use in other places as well.

File: src/status_im2/navigation/core.cljs [hunk 11]
```diff
@@ -93,131 +115,89 @@
...
+(re-frame/reg-fx :hide-bottom-sheet (fn [] (dissmiss-overlay "bottom-sheet")))
+
+;; MERGE OPTIONS
+(re-frame/reg-fx
+ :merge-options
+ (fn [{:keys [id options]}]
+   (navigation/merge-options id options)))
 
+;; LEGACY (should be removed in status 2.0)
 (defn get-screen-component
```
Comment (J-Son89) [inline src/status_im2/navigation/core.cljs:195]: how about we just rename the function `get-screen-component-DEPRECATED` or something to this effect? it's a bit more noticeable that we need to remove it then

File: src/status_im2/navigation/events.cljs [hunk 12]
```diff
@@ -56,18 +66,42 @@
...
+  {:events [:show-bottom-sheet]}
+  [{:keys [db]} content]
+  (let [{:keys [sheets hide?]} (:bottom-sheet db)]
+    (rf/merge {:db (update-in db [:bottom-sheet :sheets] #(conj % content))}
+              #(when-not hide?
+                 (if (seq sheets)
+                   (hide-bottom-sheet %)
+                   {:show-bottom-sheet nil})))))
+
+;;LEGACY
```
Comment (J-Son89) [inline src/status_im2/navigation/events.cljs]: same, for any of this Legacy pieces I would prefer that we rename them so we know they are separate. A comment in a file is not a string enough discriminator imo and it's likely it will go unnoticed for longer this way and more potential to be reused somewhere it should not be.

File: src/status_im2/navigation/options.cljs (hunk 7 above)
Comment (J-Son89) [reply src/status_im2/navigation/options.cljs:1]: Nice! :)

File: src/status_im2/navigation/roots.cljs [hunk 13]
```diff
@@ -1,78 +1,26 @@
...
   {;; ONBOARDING
    :onboarding
    {:root {:stack {:id       :onboarding
                    :children [{:component {:name    :get-your-keys
                                            :id      :get-your-keys
-                                           :options (status-bar-options)}}]
-                   :options  (merge (default-root)
-                                    (status-bar-options)
-                                    {:topBar (assoc (topbar-options)
+                                           :options (options/statusbar-and-navbar-root)}}]
```
Comment (J-Son89) [inline src/status_im2/navigation/roots.cljs:20]: consider making a separate options for all of these old options so that we do not need to think about these when adjusting the other options.
e.g options/statusbar-and-navbar-root

File: src/status_im2/navigation/core.cljs (hunk 10 above)
Comment (J-Son89) [reply src/status_im2/navigation/core.cljs:85]: also if part of this code is about managing the old screens vs new screens, we can have two functions by this and add a boolean property `old-ui` to the old screens and then use that here.
```
(let [{:keys [options old-ui]} (get views/screens comp)]
(dismiss-all-modals)
    (navigation/push
     (name @state/root-id)
     {:component {:id      comp
                  :name    comp
                  :options (if old-ui? (merge-old-options options ...) (merge-options ...)
```

File: src/status_im2/navigation/core.cljs (hunk 11 above)
Comment (ilmotta) [reply src/status_im2/navigation/core.cljs:195]: Just to add to this idea, we can also add the metadata `^:deprecated` to any var and `clj-kondo` can be configured to warn (instead of fail). Some editors also strikethrough usages of deprecated vars.

**Comment by J-Son89** [comment]
checked out your branch and it looks good. On Android we need to add to add padding-top to some screens but this is expected 👍 . Nice one @flexsurfer :)

**Review by J-Son89 — APPROVED** [review]

File: src/status_im2/navigation/core.cljs (hunk 11 above)
Comment (flexsurfer) [reply src/status_im2/navigation/core.cljs:195]: this will be removed with old code, we still merge old screens , so i could just move this into status-im namespace, no need to maintain or care about this code

File: src/status_im2/navigation/roots.cljs (hunk 13 above)
Comment (flexsurfer) [reply src/status_im2/navigation/roots.cljs:20]: what old options? we don't have old options

File: src/status_im2/navigation/core.cljs (hunk 10 above)
Comment (flexsurfer) [reply src/status_im2/navigation/core.cljs:85]: thanks, yeah i would do it in the separate PR, still lots of work to do, this PR is already huge but its only first iteration

File: src/status_im2/navigation/core.cljs (hunk 11 above)
Comment (J-Son89) [reply src/status_im2/navigation/core.cljs:195]: Yeah that's also a workable solution 👍

File: src/status_im2/navigation/roots.cljs (hunk 13 above)
Comment (J-Son89) [reply src/status_im2/navigation/roots.cljs:20]: Ok so the old screens work exactly as the new ones? I thought the designs were a bit different on the old screens and so the old options are a bit different.
For example they use the inbuilt topNav elements etc.

File: src/status_im2/navigation/core.cljs (hunk 10 above)
Comment (J-Son89) [reply src/status_im2/navigation/core.cljs:85]: Yep, sounds good 👌

File: src/js/worklets/bottom_sheet.js [hunk 14]
```diff
@@ -6,10 +6,10 @@ export function useTranslateY(initialTranslationY, bottomSheetDy, panY) {
   })
 }
 
-export function useBackgroundOpacity(translateY, backgroundHeight, windowHeight) {
+export function useBackgroundOpacity(translateY, backgroundHeight, windowHeight, opacity) {
   return useDerivedValue(() => {
-    const opacity = ((translateY.value - windowHeight) / -backgroundHeight) * 0.5
+    const calculated_opacity = ((translateY.value - windowHeight) / -backgroundHeight) * opacity
```
Comment (briansztamfater) [inline src/js/worklets/bottom_sheet.js]: Is preferable to use camelCase for variable names in JavaScript.

https://www.w3schools.com/js/js_conventions.asp

**Review by briansztamfater — APPROVED** [review]

File: src/js/worklets/bottom_sheet.js (hunk 14 above)
Comment (flexsurfer) [reply src/js/worklets/bottom_sheet.js]: good point.

**Comment by pavloburykh** [comment]
@flexsurfer thanx for the PR. Please, take a look at the following issues:

### ISSUE 1 Switching to landscape mode is enabled on Android

Device Samsung Galaxy A52. Android 12

On onboarding screens it is possible to switch to Landscape mode staring from second screen 

Inside multiaccount it is possible to switch to landscape mode after opening any bottom sheet


https://user-images.githubusercontent.com/97245802/228579975-19f710a0-7563-4d90-a741-a1d72677c610.mp4


https://user-images.githubusercontent.com/97245802/228579989-1813ef75-5be8-4974-a28e-097e3cf35416.mp4

**Comment by pavloburykh** [comment]
### ISSUE 2 Onboarding screen header elements are located within status bar (Android) 

photo_2023-03-29 17 56 31

**Comment by pavloburykh** [comment]
### ISSUE 3 Navigation bar changes mode from dark to light after opening bottom sheet

**Steps:** 

1. Open any tab, for example Messages tab
2. Open bottom sheet (tap + button)
3. Pay attention at navigation bar color

**Actual result:** Navigation bar changes mode from dark to light

photo_2023-03-29 18 00 37

https://user-images.githubusercontent.com/97245802/228581291-4e85a38a-25bf-4d02-a144-f6a2bc853c2b.mp4

**Comment by pavloburykh** [comment]
### ISSUE 4 Long pinned list/Add group members bottom sheets expand to much (IOS/Android)

For IOS it is more critical, because it is impossible to close bottom sheet by swiping down

https://user-images.githubusercontent.com/97245802/228583530-0e3fc27d-163c-48a0-93cf-1c223e74b1e8.mp4


https://user-images.githubusercontent.com/97245802/228589565-9a1c3b02-53b4-450f-8593-7da8fe7f3968.mp4



On Android it slightly overlays status bar: 

photo_2023-03-29 18 08 37

**Comment by pavloburykh** [comment]
### ISSUE 5 Keyboard opens and then closes in case of editing/replying a message (IOS)

**Steps:** 

1. Open any chat
2. Send a message
3. Longpress on message
4. Tap Edit or reply
5. Pay attention at keyboard behaviour

**Actual result:** keyboard opens and then closes


https://user-images.githubusercontent.com/97245802/228585380-b0629ba9-868f-42a8-844a-e67892c35645.mp4

**Comment by pavloburykh** [comment]
### ISSUE 6 Status bar elements' color is merged with background throughout bottom tabs (IOS)

Steps: 

1. Open any tab, for example Messages tab
2. Pay attention at status bar

**Actual result**:  status bar elements have wrong color both on light and dark modes

photo_2023-03-29 18 18 59

**Comment by pavloburykh** [comment]
### ISSUE 7 Generate QR bottom sheet is overlapped by opened keyboard (IOS)

Steps: 

1. Go to Profile - Syncing - Generate new QR
2. Proceed to password screen
3. Pay attention if keyboard overlaps bottom sheet when opened

**Actual result:** keyboard overlaps bottom sheet when opened


https://user-images.githubusercontent.com/97245802/228587825-d0058338-084a-47ea-b7d3-8b0534b8d415.mp4

**Comment by pavloburykh** [comment]
### ISSUE 8 Some screen headers are located within status bar throughout the app (IOS/Android)

Some of the screens: old community screen, Share community screen, Invite screen, Group details screen

photo_2023-03-29 18 33 18
photo_2023-03-29 18 33 20
photo_2023-03-29 18 33 24
photo_2023-03-29 18 33 28

**Comment by pavloburykh** [comment]
### ISSUE 9 "Cannot read property root of null" error and crash when creating/recovering multiaccount using **old flows**

@flexsurfer this bug can be considered to be irrelevant because old flows are to be deprecated. But could you please check if this bug does not affect new flows in anyway?  

For now I don't see any visible issues/crashes/errors related to creating/restoring account using new flows. 

**Steps:** 

1. Install app
2. Proceed with creating/restoring account using Old temporary flows
3. Observe error and crash after creating password step

**Actual result:** "Cannot read property root of null" error and crash

photo_2023-03-29 19 05 16

**Comment by flexsurfer** [comment]
wow, thats a lot of issues :( thanks @pavloburykh

File: src/status_im2/contexts/chat/photo_selector/view.cljs [hunk 15]
```diff
@@ -107,25 +107,23 @@
...
-            (rf/dispatch [:chat.ui/camera-roll-get-photos 20 nil selected-album])
-            (if (seq selected-images)
-              (reset! selected (vec (vals selected-images)))
-              (reset! selected @temporary-selected)))
-          [selected-album])
-         [bottom-sheet-screen/view
-          (fn [{:keys [scroll-enabled on-scroll]}]
+  [{:keys [scroll-enabled on-scroll]}]
+  [safe-area/consumer
+   (fn [insets]
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/photo_selector/view.cljs:113]: Please revert this change. Reading insets from consumer causes problems on this screen. Also, I checked the PR build, the photo selector screen currently keeps doing unnecessary re-renders.

File: src/status_im/ui/screens/wallet/accounts/views.cljs [hunk 16]
```diff
@@ -290,7 +291,7 @@
           [quo2.button/button
            {:type     :grey
             :size     32
-            :on-press #(re-frame/dispatch [:bottom-sheet/show-sheet
+            :on-press #(re-frame/dispatch [:bottom-sheet/show-sheet-old
```
Comment (OmarBasem) [inline src/status_im/ui/screens/wallet/accounts/views.cljs:294]: Why all these "old" sheets?

File: src/status_im/ui2/screens/chat/components/new_chat/view.cljs [hunk 17]
```diff
@@ -69,18 +64,17 @@
...
            selected-contacts                 (rf/sub [:group/selected-contacts])
-           window-height                     (rf/sub [:dimensions/window-height])
            one-contact-selected?             (= selected-contacts-count 1)
            contacts-selected?                (pos? selected-contacts-count)
            {:keys [primary-name public-key]} (when one-contact-selected?
                                                (rf/sub [:contacts/contact-by-identity
                                                         (first selected-contacts)]))
            no-contacts?                      (empty? contacts)]
-       [rn/view {:style {:height (* window-height 0.9)}}
+       [:<>
```
Comment (OmarBasem) [inline src/status_im/ui2/screens/chat/components/new_chat/view.cljs:73]: The section-list in this file would need to be changed to use a `gesture/flat-list`, and have `scroll-enabled` and `on-scroll` parameters passed to it to be able to read gestures. Please open an issue if not for this PR.

And the close button should use `close` parameter.

**Review by OmarBasem — COMMENTED** [review]
Great improvements to the bottom-sheet @flexsurfer 

UI looks much better, but there is one thing that looks a little odd. When a bottom sheet component disappears and then a bottom sheet screen appears it would feel smoother if the background overlay does not disappear and then re-appear. Maybe you can delay the bottom-sheet-component overlay disappearing and `skip-background?` of the bottom-sheet-screen, then after animation is complete, remove bottom-sheet-component background, and set `skip-background?` to false.

File: src/status_im/ui2/screens/chat/components/new_chat/view.cljs (hunk 17 above)
Comment (OmarBasem) [reply src/status_im/ui2/screens/chat/components/new_chat/view.cljs:73]: Also why not move this screen to `status-im2`

File: src/status_im2/navigation/options.cljs [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+    :layout                 {:componentBackgroundColor :transparent
+                             :orientation              :portrait
+                             :backgroundColor          :transparent}}
+   (if platform/android?
+     {:statusBar {:backgroundColor :transparent
+                  :style           :light
+                  :drawBehind      true}}
+     {:statusBar {:style :light}})))
+
+(def sheet-options
```
Comment (OmarBasem) [inline src/status_im2/navigation/options.cljs:70]: I think all sheets would have a dark status bar

**Comment by flexsurfer** [comment]
- [x] ISSUE 1 Switching to landscape mode is enabled on Android
- [ ] ~ISSUE 2~ Onboarding screen header elements are located within status bar (Android) - seems like not the issues of this PR, should be addressed separately, i've created and issue https://github.com/status-im/status-mobile/issues/15535 , please ignore this bug for now 
- [ ] ~ISSUE 3~ Navigation bar changes mode from dark to light after opening bottom sheet - known issue, will be addressed separately
- [x] SSUE 4 Long pinned list/Add group members bottom sheets expand to much (IOS/Android)
- [x] ISSUE 5 Keyboard opens and then closes in case of editing/replying a message (IOS)
- [ ] ~ISSUE 6~ Status bar elements' color is merged with background throughout bottom tabs (IOS)- known issue, will be addressed separately
- [x] ISSUE 7 Generate QR bottom sheet is overlapped by opened keyboard (IOS)
- [x] ISSUE 8 Some screen headers are located within status bar throughout the app (IOS/Android)
- [ ] ~ISSUE 9~ "Cannot read property root of null" error and crash when creating/recovering multiaccount using old flows , it was broken when new notification screen was implemented, not in this PR, there is no such root anymore -  :enable-notifications

**Comment by flexsurfer** [comment]
thank you @pavloburykh all should be fixed now, see list above

File: src/status_im/ui2/screens/chat/components/new_chat/view.cljs (hunk 17 above)
Comment (flexsurfer) [reply src/status_im/ui2/screens/chat/components/new_chat/view.cljs:73]: not moved because it must be refactored first

File: src/status_im/ui/screens/wallet/accounts/views.cljs (hunk 16 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/accounts/views.cljs:294]: because we still have old sheets in old UI, and they implemented in different designs so we have to keep old bottom sheet for now

File: src/status_im2/contexts/chat/photo_selector/view.cljs (hunk 15 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/photo_selector/view.cljs:113]: this should be fixed properly, i'll file an issue

**Comment by pavloburykh** [comment]
> thank you @pavloburykh all should be fixed now, see list above

Thanx for the fixes @flexsurfer. ISSUES 1,4,5,7,8 are fixed. 

Regarding

>ISSUE 2 Onboarding screen header elements are located within status bar (Android) - seems like not the issues of this PR, should be addressed separately, i've created and issue replace :padding-top (if platform/ios? 44 0) by using safe-area  #15535 , please ignore this bug for now

Still looks like PR issue as I am not facing it in develop. I see that current PR fixes the problem with status bar on Android, originally reported here https://github.com/status-im/status-mobile/issues/15362 But seem that fix resulted in overlaying of onboarding screen header elements on status bar. 

Regarding

>ISSUE 9 "Cannot read property root of null" error and crash when creating/recovering multiaccount using old flows , it was broken when new notification screen was implemented, not in this PR, there is no such root anymore - :enable-notifications

I am not facing this behaviour in develop so decided to log it here. Anyway if this issue does not affect new onboarding flows then we are okay to ignore it. 

@flexsurfer also, please take a look at some other issues

**Comment by pavloburykh** [comment]
### ISSUE 10 Data is not scrollable within bottom sheets (Android )

Examples of bottom sheets that are expected to be scrollable: long pinned list, community rules, new chat contact list, manage group members contact list


https://user-images.githubusercontent.com/97245802/228833335-c14b91a6-8e8a-46af-8424-5148fb6ade5c.mp4


https://user-images.githubusercontent.com/97245802/228833344-3d2d1cd1-29ad-4dd0-aa7c-d1f45ddbb9ad.mp4


https://user-images.githubusercontent.com/97245802/228833349-3c4c77e5-1558-4e94-8ac2-b9e183fe1843.mp4

**Comment by pavloburykh** [comment]
### ISSUE 11 Confirm selection button is dislocated to the top of select image bottom sheet (Android) 

**Expected result:** Confirm button should be located at the bottom of the screen

**Actual result:** 

photo_2023-03-30 15 18 36

**Comment by pavloburykh** [comment]
### ISSUE 12 Sync QR scanner is dislocated and overlays elements above

This one might be related to [ISSUE 2](https://github.com/status-im/status-mobile/pull/15467#issuecomment-1488854991)

photo_2023-03-30 15 20 21

**Comment by flexsurfer** [comment]
> Still looks like PR issue as I am not facing it in develop. I see that current PR fixes the problem with status bar on Android,

there was a hidden but, now when we don't have it, issues are visible, but they should be fixed separately, because they should be implemented individually in each screen and ISSUE 12 from this category, should be fixed separetely

**Comment by flexsurfer** [comment]
> I am not facing this behaviour in develop so decided to log it here.

mhm i'll take a look probably it was a merge conflict then, but still only this one affected

**Comment by pavloburykh** [comment]
@flexsurfer also, could you please rebase the PR to latest develop. There are important commits which need to be included in order we be able to run e2e in current PR. Thanx!

**Comment by flexsurfer** [comment]
thanks @pavloburykh  10 and 11 should be fixed now

**Comment by pavloburykh** [comment]
@flexsurfer thanx for the fixes. [Issue 11](https://github.com/status-im/status-mobile/pull/15467#issuecomment-1490209778) is fixed. 

Issue 10 is fixed partially: Currently I can scroll within pinned list, new chat contact list. 

But still cannot scroll over Manage group contact list (see video below) 

https://user-images.githubusercontent.com/97245802/228909031-6efefc58-7568-4c0d-986e-c030e8760119.mp4

Also, there is still an issue with Request community join screen: checkbox and the buttons (Cancel/Request join) are hidden. No way to scroll to them. 

photo_2023-03-30 19 56 38

BTW, comparing to develop build, checkbox/buttons are visible on my device (Samsung Galaxy A52, Android 12) even with no scroll because bottom sheet expands higher

photo_2023-03-30 20 05 24

**Comment by pavloburykh** [comment]
### ISSUE 13 User gets stuck on Syncing device screen after scanning sync QR code

Not reproducing the bug in latest develop

**Steps:** 

1. User A generates sync QR (Profile => Syncing => Sync new device)
2. User B Scans QR code (select Sign in option during onboarding) 
3. See if User B redirected to Enable notifications screen after Syncing device.. screen 

**Actual result:** User gets stuck on Syncing device screen after scanning sync QR code 


https://user-images.githubusercontent.com/97245802/228914357-e5adf176-ea94-4c33-b87d-80a961b949bc.mp4



**Expected result:** User is redirected to Enable notif screen after syncing is finished. Recorded on latest develop build. 

https://user-images.githubusercontent.com/97245802/228913805-3f43229b-f156-4d9e-9868-5d600567b4c4.mp4

**Comment by pavloburykh** [comment]
### ISSUE 14 Error after opening and closing device's notification bar while chat screen was opened

This one has been caught by e2e. Not able reproduce it manually yet. Below I attach logs and video from saucelab. 

Here is [video](https://app.eu-central-1.saucelabs.com/tests/972b3d064af24146aa41dc788a4e3bd0?auth=e11eb57e73241b566819758cd19a677c#350)

[logcat (3).log](https://github.com/status-im/status-mobile/files/11114305/logcat.3.log)

Test Results | Test Details | Sauce Labs 2023-03-30 20-30-59

File: src/status_im2/contexts/chat/photo_selector/view.cljs (hunk 15 above)
Comment (OmarBasem) [reply src/status_im2/contexts/chat/photo_selector/view.cljs:113]: @flexsurfer why don't we change the way we use insets. Since we use custom navigation bars, that means all screens need to use insets, and even many components. Reading insets from `safe-area/consumer` has mainly 2 problems: 

1. On Android, gives inaccurate big values for the first few renders 
2. Causes unnecessary re-renders (and even in some cases I faced crashes because of safe-area). 

Since safe area values are a constant, how about we read them on one of the initial screens, and then store them globally in an atom / app-db / localstorage?

Comment (flexsurfer) [reply src/status_im2/contexts/chat/photo_selector/view.cljs:113]: yes thats the plan

Comment (flexsurfer) [reply src/status_im2/contexts/chat/photo_selector/view.cljs:113]: https://github.com/status-im/status-mobile/issues/15535

**Comment by flexsurfer** [comment]
- [x] But still cannot scroll over Manage group contact list 
- [x] ISSUE 13 User gets stuck on Syncing device screen after scanning sync QR code
- [x] ISSUE 14 Error after opening and closing device's notification bar while chat screen was opened

thank you @pavloburykh , all should be fixed now

**Comment by pavloburykh** [comment]
@flexsurfer thanx for the fixes! 

Request community access and Manage group members bottom sheets are broken. Could you please take a look? 


https://user-images.githubusercontent.com/97245802/229080048-8d40637e-1bbe-4278-99d2-b7bb55a6e6f9.mp4


https://user-images.githubusercontent.com/97245802/229080070-aabb6c31-688b-452c-bca2-fe7898a57b69.mp4

**Comment by pavloburykh** [comment]
> @flexsurfer thanx for the fixes!
> 
> Request community access and Manage group members bottom sheets are broken. Could you please take a look?

On IOS Request community access bottom sheet is located to high from the bottom. 


https://user-images.githubusercontent.com/97245802/229083044-5357ea4d-2c6f-4f53-a447-bc9b924e5b57.mp4



Also, Cancel button on Manage group members screen does not respond both on IOS and Android


https://user-images.githubusercontent.com/97245802/229083064-37b90098-c1c9-4d6b-a7ce-7a4484b4d10f.mp4

**Comment by flexsurfer** [comment]
thanks @pavloburykh should be fixed

**Comment by churik** [comment]
@flexsurfer please, make sure you've pulled my commit until pushing new fixes, thanks!

**Comment by churik** [comment]
@flexsurfer FYI: one more commit, mind when squashing

**Comment by pavloburykh** [comment]
@flexsurfer thank you for your work! PR is tested and ready to be merged.
